packages/web-components/src/components/ic-tooltip/ic-tooltip.tsx

```diff
@@ -123,9 +105,111 @@ export class Tooltip {
     show ? this.show() : this.hide();
   }
 
+  private getTooltipTranslate = (dialogEl: DOMRect) => {
+    const child = this.el.children[0].getBoundingClientRect();
+    let tooltipX;
+    let tooltipY;
+    switch (this.placement) {
+      case "bottom":
+        tooltipX = child.left - dialogEl.left - 0.5 * child.width;
+        tooltipY = child.bottom - dialogEl.top;
+        break;
+      case "bottom-start":
+        tooltipX = child.left - dialogEl.left;
+        tooltipY = child.bottom - dialogEl.top;
+        break;
+      case "bottom-end":
+        tooltipX = child.right - dialogEl.right;
+        tooltipY = child.bottom - dialogEl.top;
+        break;
+      case "top":
+        tooltipX = child.left - dialogEl.left - 0.5 * child.width;
+        tooltipY = child.top - dialogEl.bottom;
+        break;
+      case "top-start":
+        tooltipX = child.left - dialogEl.left;
+        tooltipY = child.top - dialogEl.bottom;
+        break;
+      case "top-end":
+        tooltipX = child.right - dialogEl.right;
+        tooltipY = child.top - dialogEl.bottom;
+        break;
+      case "left":
+      case "left-start":
+        tooltipX = child.right - dialogEl.right - child.width;
+        tooltipY = child.bottom - dialogEl.top - child.height;
+        break;
+      case "left-end":
+        tooltipX = child.right - dialogEl.right - child.width;
+        tooltipY = child.top - dialogEl.bottom + child.height;
+        break;
+      case "right":
+      case "right-start":
+        tooltipX = child.left - dialogEl.left + child.width;
+        tooltipY = child.bottom - dialogEl.top - child.height;
+        break;
+      case "right-end":
+        tooltipX = child.left - dialogEl.left + child.width;
+        tooltipY = child.top - dialogEl.bottom + child.height;
+        break;
+    }
+    if (this.dialogOverflow && tooltipX < 0) {
+      if (this.placement.includes("top") || this.placement.includes("bottom")) {
+        this.toolTip.style.setProperty(
+          "--tooltip-arrow-translate",
+          `${tooltipX}px`
+        );
+        tooltipX = child.left - dialogEl.left;
+      }
+      if (this.placement.includes("left")) {
+        this.placement = "right";
+        tooltipX = child.left - dialogEl.left + child.width;
+      }
+    }
+
+    this.toolTip.style.setProperty("--tooltip-translate-x", `${tooltipX}px`);
+    this.toolTip.style.setProperty("--tooltip-translate-y", `${tooltipY}px`);
+  };
+
   private show = () => {
     this.toolTip.setAttribute("data-show", "");
-    this.popperInstance.update();
+
+    this.icDialogEl = this.el.closest("ic-dialog");
```

Comment:
think there is scope for tidying up this code, so it is not run every time the tooltip is shown. Could determine if it is on a dialog in the `componentWillLoad `or `componentDidLoad`.

the `on-dialog` class could then be applied conditionally in the render